packages/cli/src/checks/INCORRECT_REPOSITORY_FIELD.ts

```diff
@@ -0,0 +1,71 @@
+import parseGithubUrl from "parse-github-url";
+import path from "path";
+import normalizePath from "normalize-path";
+import { makeCheck, Workspace } from "./utils";
+
+type ErrorType = {
+  type: "INCORRECT_REPOSITORY_FIELD";
+  workspace: Workspace;
+  currentRepositoryField: string | undefined;
+  correctRepositoryField: string;
+};
+
+export default makeCheck<ErrorType>({
+  type: "all",
+  validate: (workspace, allWorkspaces, rootWorkspace) => {
+    let rootRepositoryField = (rootWorkspace.config as any).repository;
+
+    if (typeof rootRepositoryField === "string") {
+      let result = parseGithubUrl(rootRepositoryField);
+      if (result !== null) {
+        if (workspace === rootWorkspace) {
+          let correctRepositoryField = `https://github.com/${result.owner}/${result.name}`;
```

Comment:
Just a thing, this makes manypkg exclusively for GH, which I don't think is ideal. Would love to make this more generic.

I'd also love a bit more flexibility in defining this, as this limits the spec (I personally quite like the

```
"repository": {
  "type" : "git",
  "url" : "https://github.com/facebook/react.git",
  "directory": "packages/react-dom"
}
```

format for this)

This comment really relates to further down, but would love to see if an object was used in the root, we recommended and fixed to an object.

Everything else looks good, some nice tidying done to tests and types.

Reply:
> Just a thing, this makes manypkg exclusively for GH, which I don't think is ideal. Would love to make this more generic.

💯. A little note with this: This doesn't make Manypkg exclusively for GH, it only makes this rule exclusive to GH

---

I specifically don't want to use objects because the directory field doesn't do anything as far as I can see (e.g. if you click the repository link on https://www.npmjs.com/package/react-dom, it goes to the root, not into the directory)